### PR TITLE
[SRVKP-6366] Fixed usage of SUBDIRECTORY

### PR DESCRIPTION
**Pull request by Aneesh-M-Bhat**
- fixed SUBDIRECTORY having wrong behaviour and added param to tests as well.
- Removed ls from load-scripts.

ISSUE: https://issues.redhat.com/browse/SRVKP-6366

### CODE REVIEW

File: test/e2e/resources/pipeline-maven.yaml [hunk 1]
```diff
@@ -48,6 +47,8 @@ spec:
           value: "false"
         - name: VERBOSE
           value: "$(params.VERBOSE)"
+        - name: SUBDIRECTORY
+          value: $(context.pipelineRun.name)
```
Comment (savitaashture) [inline test/e2e/resources/pipeline-maven.yaml:51]: what if subdirectory is npt given will the woking behavior is same ?

Comment (Aneesh-M-Bhat) [reply test/e2e/resources/pipeline-maven.yaml:51]: yes it is, checked that as well, i felt that our tests should have all params so ya, could include 2 tests if u feel thats needed, but yes i did check them out, it works as previously when Subdirectory is not populated.

**Comment by Aneesh-M-Bhat** [comment]
```
---
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  labels:
    name: task-maven
  name: task-maven
spec:
  workspaces:
    - name: source
      optional: false
    - name: server_secret
      optional: true
    - name: proxy_secret
      optional: true
    - name: proxy_configmap
      optional: true

  params:
    - name: URL
      type: string
    - name: REVISION
      type: string
    - name: VERBOSE
      type: string

  tasks:
    - name: git-clone
      taskRef:
        resolver: git
        params:
          - name: url
            value: https://github.com/openshift-pipelines/tektoncd-catalog
          - name: revision
            value: p
          - name: pathInRepo
            value: tasks/task-git-clone/0.3.0/task-git-clone.yaml
      workspaces:
        - name: output
          workspace: source
      params:
        - name: URL
          value: "$(params.URL)"
        - name: REVISION
          value: "$(params.REVISION)"
        - name: SUBMODULES
          value: "false"
        - name: VERBOSE
          value: "$(params.VERBOSE)"
        - name: SUBDIRECTORY
          value: $(context.pipelineRun.name)
    - name: maven
      taskRef:
        name: maven
      params:
        - name: SERVER_SECRET
          value: "$(params.SERVER_SECRET)"
        - name: GOALS
          value:
            - "validate"
        - name: SUBDIRECTORY
          value: $(context.pipelineRun.name)
      runAfter:
        - git-clone
      workspaces:
        - name: source
          workspace: source
```
image

& tested after removing subdirectory from the above pipeline
ran smoothly 
image

i did some logging to confirm that the path are also correct & aren't the smae in the both the contexts so it should be good to merge.

**Review by savitaashture — APPROVED** [review]
/lgtm

**Pull request merged**